include/cuda/std/detail/libcxx/include/__expected/expected_base.h

```diff
@@ -0,0 +1,1016 @@
+    __with_val.__has_val_ = false;
+    __with_err.__has_val_ = true;
+  }
+};
+
+template <class _Tp, class _Err, bool =
+          (_LIBCUDACXX_TRAIT(is_trivially_copy_constructible, _Tp) || _LIBCUDACXX_TRAIT(is_same, _Tp, void)) &&
+           _LIBCUDACXX_TRAIT(is_trivially_copy_constructible, _Err)>
+struct __expected_copy : __expected_storage<_Tp, _Err>
+{
+  using __base = __expected_storage<_Tp, _Err>;
+#if defined(_LIBCUDACXX_COMPILER_NVRTC) // nvbug3961621
+  constexpr __expected_copy() noexcept = default;
+
+  template<class... _Args>
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr
+  __expected_copy(_Args&&... __args) noexcept(noexcept(__base(cuda::std::declval<_Args>()...)))
+    : __base(_CUDA_VSTD::forward<_Args>(__args)...)
+  {}
+#else // ^^^ _LIBCUDACXX_COMPILER_NVRTC ^^^ / vvv !_LIBCUDACXX_COMPILER_NVRTC vvv
+  using __base::__base;
+#endif // !_LIBCUDACXX_COMPILER_NVRTC
+};
+
+template <class _Tp, class _Err>
+struct __expected_copy<_Tp, _Err, false> : __expected_storage<_Tp, _Err>
+{
+  using __base = __expected_storage<_Tp, _Err>;
+
+#if defined(_LIBCUDACXX_COMPILER_NVRTC) // nvbug3961621
+  template<class... _Args>
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr
+  __expected_copy(_Args&&... __args) noexcept(noexcept(__base(cuda::std::declval<_Args>()...)))
+    : __base(_CUDA_VSTD::forward<_Args>(__args)...)
+  {}
+#else // ^^^ _LIBCUDACXX_COMPILER_NVRTC ^^^ / vvv !_LIBCUDACXX_COMPILER_NVRTC vvv
+  using __base::__base;
+#endif // !_LIBCUDACXX_COMPILER_NVRTC
+
+  __expected_copy() noexcept = default;
+
+  _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_CONSTEXPR_AFTER_CXX17
+  __expected_copy(const __expected_copy& __other)
+    noexcept(_LIBCUDACXX_TRAIT(is_nothrow_copy_constructible, _Tp)
+          && _LIBCUDACXX_TRAIT(is_nothrow_copy_constructible, _Err))
+    : __base(__other.__has_val_)
+  {
+    if (__other.__has_val_) {
+      _CUDA_VSTD::__construct_at(_CUDA_VSTD::addressof(this->__union_.__val_), __other.__union_.__val_);
+    } else {
+      _CUDA_VSTD::__construct_at(_CUDA_VSTD::addressof(this->__union_.__unex_), __other.__union_.__unex_);
+    }
+  }
+
+  __expected_copy(__expected_copy&&) = default;
+  __expected_copy& operator=(const __expected_copy&) = default;
+  __expected_copy& operator=(__expected_copy&&) = default;
+};
+
+template <class _Tp, class _Err, bool =
+          (_LIBCUDACXX_TRAIT(is_trivially_move_constructible, _Tp) || _LIBCUDACXX_TRAIT(is_same, _Tp, void)) &&
+           _LIBCUDACXX_TRAIT(is_trivially_move_constructible, _Err)>
+struct __expected_move : __expected_copy<_Tp, _Err>
+{
+  using __base = __expected_copy<_Tp, _Err>;
+#if defined(_LIBCUDACXX_COMPILER_NVRTC) // nvbug3961621
+  constexpr __expected_move() noexcept = default;
+
+  template<class... _Args>
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr
+  __expected_move(_Args&&... __args) noexcept(noexcept(__base(cuda::std::declval<_Args>()...)))
+    : __base(_CUDA_VSTD::forward<_Args>(__args)...)
```

Comment:
Unify the qualification in these, probably both to `_CUDA_VSTD`, since this is in a header?

Reply:
Great catch, fixed